### ISSUE TITLE
not installable on M1

### ISSUE THREAD

**Issue by yarikoptic** [conda-forge/nipype-feedstock#99]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

it seems that nipype package is arch specific according to https://anaconda.org/conda-forge/nipype . As a result -- it is not installable on M1

```
(heudiconv) datalad@26696 ~ % conda install -c conda-forge -y nipype   
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - nipype

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-arm64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/osx-arm64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-arm64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.

```

which in case of installing downstream heudiconv project results even in worse behavior of noninformative empty message:  https://github.com/conda/conda/issues/10672#issuecomment-1730238942

So, likely just needs m1 build or announce package a noarch altogether. (in datalad we switch from noarch to arch specific only to run tests across different python versions... was that a motivation here or there are any extensions?)

attn @effigies

### Installed packages

```shell
(heudiconv) datalad@26696 ~ % conda list                            
# packages in environment at /Users/datalad/miniconda-test3/envs/heudiconv:
#
# Name                    Version                   Build  Channel
bz2file                   0.98                       py_0    conda-forge
bzip2                     1.0.8                h3422bc3_4    conda-forge
ca-certificates           2023.7.22            hf0a4a13_0    conda-forge
dcm2niix                  1.0.20230411         h1995070_0    conda-forge
dcmstack                  0.8.0              pyhd8ed1ab_2    conda-forge
importlib-metadata        6.8.0              pyha770c72_0    conda-forge
isodate                   0.6.1              pyhd8ed1ab_0    conda-forge
libblas                   3.9.0           18_osxarm64_openblas    conda-forge
libcblas                  3.9.0           18_osxarm64_openblas    conda-forge
libcxx                    16.0.6               h4653b0c_0    conda-forge
libexpat                  2.5.0                hb7217d7_1    conda-forge
libffi                    3.4.2                h3422bc3_5    conda-forge
libgfortran               5.0.0           13_2_0_hd922786_1    conda-forge
libgfortran5              13.2.0               hf226fd6_1    conda-forge
liblapack                 3.9.0           18_osxarm64_openblas    conda-forge
libopenblas               0.3.24          openmp_hd76b1f2_0    conda-forge
libsqlite                 3.43.0               hb31c410_0    conda-forge
libzlib                   1.2.13               h53f4e23_5    conda-forge
llvm-openmp               16.0.6               h1c12783_0    conda-forge
ncurses                   6.4                  h7ea286d_0    conda-forge
nibabel                   5.0.1              pyhd8ed1ab_0    conda-forge
numpy                     1.26.0          py311hb8f3215_0    conda-forge
openssl                   3.1.3                h53f4e23_0    conda-forge
packaging                 23.1               pyhd8ed1ab_0    conda-forge
pathlib                   1.0.1                      py_1    conda-forge
pip                       23.2.1             pyhd8ed1ab_0    conda-forge
pydicom                   2.4.3              pyhd8ed1ab_0    conda-forge
pyparsing                 3.1.1              pyhd8ed1ab_0    conda-forge
python                    3.11.5          h47c9636_0_cpython    conda-forge
python_abi                3.11                    4_cp311    conda-forge
rdflib                    7.0.0              pyhd8ed1ab_0    conda-forge
readline                  8.2                  h92ec313_1    conda-forge
setuptools                68.2.2             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
tk                        8.6.12               he1e0b03_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
wheel                     0.41.2             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h57fd34a_0    conda-forge
zipp                      3.17.0             pyhd8ed1ab_0    conda-forge
```

### Environment info

```shell
(heudiconv) datalad@26696 ~ % conda info

     active environment : heudiconv
    active env location : /Users/datalad/miniconda-test3/envs/heudiconv
            shell level : 1
       user config file : /Users/datalad/.condarc
 populated config files : /Users/datalad/.condarc
          conda version : 4.12.0
    conda-build version : not installed
         python version : 3.9.18.final.0
       virtual packages : __osx=12.1=0
                          __unix=0=0
                          __archspec=1=arm64
       base environment : /Users/datalad/miniconda-test3  (writable)
      conda av data dir : /Users/datalad/miniconda-test3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-arm64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-arm64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/datalad/miniconda-test3/pkgs
                          /Users/datalad/.conda/pkgs
       envs directories : /Users/datalad/miniconda-test3/envs
                          /Users/datalad/.conda/envs
               platform : osx-arm64
             user-agent : conda/4.12.0 requests/2.31.0 CPython/3.9.18 Darwin/21.2.0 OSX/12.1
                UID:GID : 502:20
             netrc file : None
           offline mode : False
```

**Comment by effigies** [comment]
Some dependencies are windows only, and conda prevents those in noarch packages.

**Comment by yarikoptic** [comment]
Eh, despite that I don't see anything on that in https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#architecture-independent-packages  I recalled our attempt in datalad : https://github.com/conda-forge/datalad-feedstock/pull/105#issuecomment-1538816417 

`noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.

so indeed so ...  Then I guess should be scheduled for M1 rebuild. I forgot which bot to ask what to make it happen... I think this is it: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4934

**Comment by effigies** [comment]
Should be installable on M1 once the build finishes. It won't be installable on Python 3.11, though. Nipype is capped at 3.10 due to major breaking changes in traits 6.4, and 6.3 only shipped wheels up to 3.10. Short of a significant rewrite of nipype, its end-of-life will coincide with Python 3.10.
